src/compiler/utilities.ts

```diff
@@ -5274,6 +5280,10 @@ namespace ts {
 
     /* @internal */
     export function isDeclaration(node: Node): node is NamedDeclaration {
+        if (node.kind === SyntaxKind.TypeParameter) {

```

Comment:
In a TypeScript file, a TypeParameter in JSDoc is just a comment, albeit one that is parsed.
In a JavaScript file, it's actually a declaration.